### PR TITLE
Scoutgame telegram onboarding

### PR DESCRIPTION
**Pull request by Devorein**

### CODE REVIEW

File: apps/scoutgametelegram/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "typecheck": "../../node_modules/typescript/bin/tsc --project ./tsconfig.json  --noEmit"
   },
   "dependencies": {
+    "@hookform/resolvers": "^3.9.1",
```
Comment (valentinludu) [inline apps/scoutgametelegram/package.json:18]: I see we already have it in the main pacakge.json

File: apps/scoutgametelegram/lib/session/loadUserAction.ts [hunk 2]
```diff
@@ -23,12 +22,8 @@ export const loadUser = actionClient
     if (!validatedData.user?.id) {
       throw new DataNotFoundError('No telegram user id found');
     }
-    const avatar = await getUserAvatar(validatedData.user.id).catch((error) => {
```
Comment (valentinludu) [inline apps/scoutgametelegram/lib/session/loadUserAction.ts:26]: We need the avatar here. I just forgot to spread it in the next line
`findOrCreateTelegramUser({ ...validatedData.user, photo_url: avatar })) as unknown as SessionUser;`

File: apps/scoutgametelegram/hooks/useInitTelegramData.ts [hunk 3]
```diff
@@ -9,19 +9,13 @@ import { useIsMounted } from 'hooks/useIsMounted';
...
     onSuccess: async (data) => {
       if (data.data) {
-        await refreshUser();
-        // @TODO: Enable this when we will have an onboarding page
-        // if (data.data?.onboardedAt && data.data?.agreedToTermsAt) {
-        //   redirect('/home');
-        // } else {
-        //   redirect('/welcome/onboarding');
-        // }
+        await refreshUser(data.data);
```
Comment (valentinludu) [inline apps/scoutgametelegram/hooks/useInitTelegramData.ts:18]: data returns just a partial user not a SessionUser event tho  you are asserting it in the loadUserAction. Better to leave it as refreshUser() and not assert it or return the whole SessionUser in loadUserAction.

File: apps/scoutgametelegram/hooks/useInitTelegramData.ts [hunk 4]
```diff
@@ -9,19 +9,13 @@ import { useIsMounted } from 'hooks/useIsMounted';
 import { loadUser } from 'lib/session/loadUserAction';
 
 export function useInitTelegramData() {
-  const telegramInitData = WebApp.initData;
+  const telegramInitData = typeof window !== 'undefined' ? WebApp.initData : null;
```
Comment (valentinludu) [inline apps/scoutgametelegram/hooks/useInitTelegramData.ts:12]: Real nice! I saw the errors this morning.

File: apps/scoutgame/components/home/HomePage.tsx [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { Box, Stack, Typography, Container } from '@mui/material';
+import { Box, Stack, Typography } from '@mui/material';
```
Comment (valentinludu) [inline apps/scoutgame/components/home/HomePage.tsx:1]: This is Scoutgame app. Are you sure you need to modify this file?

File: apps/scoutgametelegram/components/layout/PageContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { Box } from '@mui/material';
+
+import { Header } from 'components/common/Header';
+import { InfoBackgroundImage } from 'components/layout/InfoBackgroundImage';
+import { StickyFooter } from 'components/layout/StickyFooter';
+
+export function PageContainer({ children }: { children: React.ReactNode }) {
```
Comment (valentinludu) [inline apps/scoutgametelegram/components/layout/PageContainer.tsx:7]: This should not be a file. It should be like in scoutgame using a  route group with a different layout.
Pages under a certain route group should have the same layout.
https://nextjs.org/docs/app/building-your-application/routing/route-groups

File: apps/scoutgame/components/home/HomePage.tsx (hunk 5 above)
Comment (Devorein) [reply apps/scoutgame/components/home/HomePage.tsx:1]: this required a fix. I pushed it to prod on a separate PR.

**Pull request merged**